rustls/src/msgs/deframer.rs

```diff
@@ -70,12 +70,22 @@ impl MessageDeframer {
             let mut rd = codec::Reader::init(&self.buf[start..self.used]);
             let m = match OpaqueMessage::read(&mut rd) {
                 Ok(m) => m,
-                Err(MessageError::TooShortForHeader | MessageError::TooShortForLength) => {
-                    return Ok(None)
-                }
-                Err(_) => {
+                Err(msg_err) => {
```

Comment:
Why change this to move the `TooShortForHeader | TooShortForLength` arms into a nested `match`? That doesn't seem like an improvement to me, since they're handled differently than all the the other cases.

Hmm, I guess it has to do with the exhaustiveness of matching here? I guess it's a little ugly but I can't quite think of a better solution.

Reply:
Yes, that's the reason why 👍. There's no way to tell Rust you handled some cases in a different early return branch and I figured this was more preferable to using `unreachable!()`.

Reply:
Okay, let's leave it like this, I guess.